src/DynamoCore/Extensions/ExtensionCommandExecutive.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dynamo.Models;
+
+namespace Dynamo.Extensions
+{
+    internal class ExtensionCommandExecutive : ICommandExecutive
+    {
+        private readonly DynamoModel dynamoModel;
+
+        public ExtensionCommandExecutive(DynamoModel model)
+        {
+            dynamoModel = model;
+        }
+
+        public void ExecuteCommand(DynamoModel.RecordableCommand command)
+        {
+            // log that the command is being executed
+            if (dynamoModel.DebugSettings.VerboseLogging)
+            {
+                dynamoModel.Logger.Log("Command: " + command);
+            }
```

Comment:
Consider logging a less verbose message some thing like:

```
"ExtensionCommandExecutive ( UniqueId: id, Name: name, commandTypeName: AddNodeCommand )"
```

Good logging will likely greatly help in the future.